routes/auth.routes.js

```diff
@@ -0,0 +1,101 @@
+const router = require('express').Router();
+const bcrypt = require('bcryptjs');
+const User = require('../models/User.model');
+const mongoose = require('mongoose');
+const { isLoggedIn, isLoggedOut } = require('../middleware/route-guard');
+
+router.get('/signup', isLoggedOut, (req, res) => res.render('auth/signup'));
+
+router.post('/signup', async (req, res, next) => {
+    const { username, password } = req.body;
+  
+    try {
+      if (!username || !password) {
+        res.render('auth/signup', {
+          errorMessage: 'All the fields are mandatory. Please input an username and passowrd',
+        });
+        return;
+      }
+      
+      const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6}/;
+      if (!regex.test(password)) {
+        res.status(500).render('auth/signup', {
+          errorMessage:
+            'Invalid password, password needs to have at least 6 characters and include an uppercase and lowercase character',
+        });
+      }
+  
+      const salt = await bcrypt.genSalt(10);
+  
+      const hashedPassword = await bcrypt.hash(password, salt);
+  
+      const createdUser = await User.create({ username, password: hashedPassword });
+  
+      res.redirect('/');
```

Comment:
Here you could've redirected to the profile already.

`res.redirect('/profile');`